beetsplug/audible.py

```diff
@@ -104,11 +248,117 @@ def __init__(self):
         self.add_media_field("subtitle", subtitle)
 
     def album_distance(self, items, album_info, mapping):
-        dist = get_distance(data_source=self.data_source, info=album_info, config=self.config)
+        dist = beets.autotag.hooks.Distance()
         return dist
 
     def track_distance(self, item, track_info):
-        return get_distance(data_source=self.data_source, info=track_info, config=self.config)
+        dist = beets.autotag.hooks.Distance()
+        dist.add_string("track_title", item.title, track_info.title)
+        return dist
+
+    def attempt_match_trust_source_numbering(self, items: List[Item], album: AlbumInfo) -> Optional[List[Item]]:
+        """If the input album is numbered and the number range is contiguous (doesn't skip any numbers), then trust
+        that and start the index from 1 if it's not already."""
+        sorted_tracks = sorted(items, key=lambda t: t.track)
+        if is_continuous_number_series([t.track for t in sorted_tracks]):
+            # if the track is zero indexed, re-number them
+            if sorted_tracks[0].track != 1:
+                matches = []
+                for i, item in enumerate(sorted_tracks, start=1):
```

Comment:
As below, that's not what this does, it just starts i at 1 instead of zero. The incrementation of i and the for loop of the items are actually unrelated in terms of list slices.

Reply:
Ah, I get what you mean, it just sets where the returned index starts but still iterates over the whole list.

Reply:
Just so.